### ISSUE TITLE
Windows LibTorch C++ Cannot Load CUDA Module

### ISSUE THREAD

**Issue by HapeMask** [pytorch/pytorch#24793]
## 🐛 Windows LibTorch C++ Cannot Load CUDA Module

`th.jit.trace()`'ing or `th.jit.script()`'ing even just a Conv2d module will produce a serialized file that cannot be loaded by the C++ runtime. See the following two example scripts for a repro:

```python
import torch as th
from torch import nn

d = th.nn.Conv2d(3, 3, 1, 1, 0).cuda()
z = th.zeros(1, 3, 256, 256).cuda()
th.jit.script(d).save("smod.ptj") # Also fails if this is th.jit.trace(d, (z,))
```

```c++
#include <iostream>
#include <torch/script.h>
int main() {
    try {
        auto mod = torch::jit::load("smod.ptj");
    } catch(c10::Error& e) {
        std::cerr << e.what() << std::endl;
        return 1;
    }
    return 0;
}
```

## To Reproduce

1. Run the above python script.
1. Compile + run the above C++ program.

The exception text is:

```
cuda_dispatch_ptr INTERNAL ASSERT FAILED at C:\w\1\s\windows\pytorch\aten\src\ATen/native/DispatchStub.h:70, please report a bug to PyTorch. DispatchStub: missing CUDA kernel (operator () at C:\w\1\s\windows\pytorch\aten\src\ATen/native/DispatchStub.h:70)
(no backtrace available)
```

and is thrown at `include\ATen\core\TensorMethods.h:249` (the end of an inline `copy_` method).

Problem persists with latest master builds as well.

I used linux to dump the module since installing pytorch from anaconda currently results in a pytorch that will not import:
```
    from torch._C import *
ImportError: DLL load failed: The specified procedure could not be found.
```

## Expected behavior
C++ program returns 0 w/no errors displayed.

## Environment

 - PyTorch Version (e.g., 1.0): 1.2 stable (also nightly)
 - OS (e.g., Linux): Windows
 - How you installed PyTorch (`conda`, `pip`, source): conda (linux), pytorch.org (win binaries)
 - Build command you used (if compiling from source):
 - Python version: 3.7
 - CUDA/cuDNN version: 10.0 / 3.6
 - GPU models and configuration: Titan RTX

cc @peterjc123

**Comment by HapeMask** [comment]
Sorry it looks like this is a Windows-specific issue, closing til I find out more details and will re-open with a better example.

**Comment by HapeMask** [comment]
I realize the fact that I had to dump the module on linux and load on windows may confuse the issue but this is our typical use-case. Once I can import torch on windows in my anaconda distribution there I'll re-run the test and see if it still fails.

**Comment by peterjc123** [comment]
About the exception, maybe https://github.com/pytorch/pytorch/issues/22681#issuecomment-521818007 is related.
About the dll loading error, you could debug with https://github.com/lucasg/Dependencies.

**Comment by HapeMask** [comment]
@peterjc123 ahh Thanks! I originally had the crash in release mode without the assert so I switched to the debug build to debug, that may have confused things further. I’ll switch back to release and see if I can’t find the crash somehow without the debug build.

To be clear, there was an exception (c10::Error) thrown in release mode in the load call but if I tried to catch it and print e.what(), another exception would be thrown somewhere and I couldn’t figure out why so I switched to debug.
